lib/logstash/filters/mutate.rb

```diff
@@ -347,95 +347,96 @@ def uppercase(event)
         end
+      when String
+        # nil means no change was made to the String
+        original.upcase! || original
+      else
+        @logger.debug? && @logger.debug("Can't uppercase something that isn't a string", :field => field, :value => original)
+        original
+      end
     end
-  end # def uppercase
+  end
 
-  private
   def lowercase(event)
     #see comments for #uppercase
     @lowercase.each do |field|
       original = event[field]
       event[field] = case original
-        when Array
-          original.map! do |elem|
-            (elem.is_a?(String) ? elem.downcase : elem)
-          end
-        when String
-          original.downcase! || original
-        else
-          @logger.debug("Can't lowercase something that isn't a string",
-                        :field => field, :value => original)
-          original
+      when Array
+        original.map! do |elem|
+          (elem.is_a?(String) ? elem.downcase : elem)
         end
+      when String
+        original.downcase! || original
+      else
+        @logger.debug? && @logger.debug("Can't lowercase something that isn't a string", :field => field, :value => original)
+        original
+      end
     end
-  end # def lowercase
-
+  end
 
-  private
   def split(event)
     @split.each do |field, separator|
-      if event[field].is_a?(String)
-        event[field] = event[field].split(separator)
+      value = event[field]
+      if value.is_a?(String)
+        event[field] = value.split(separator)
       else
-        @logger.debug("Can't split something that isn't a string",
-                      :field => field, :value => event[field])
+        @logger.debug? && @logger.debug("Can't split something that isn't a string", :field => field, :value => event[field])
       end
     end
   end
 
-  private
   def join(event)
     @join.each do |field, separator|
-      if event[field].is_a?(Array)
-        event[field] = event[field].join(separator)
+      value = event[field]
+      if value.is_a?(Array)
+        event[field] = value.join(separator)
       end
     end
   end
 
-  private
   def strip(event)
     @strip.each do |field|
-      if event[field].is_a?(Array)
-        event[field] = event[field].map{|s| s.strip }
-      elsif event[field].is_a?(String)
-        event[field] = event[field].strip
+      value = event[field]
+      case value
+      when Array
+        event[field] = value.map{|s| s.strip }
+      when String
+        event[field] = value.strip
       end
     end
   end
 
-  private
   def merge(event)
     @merge.each do |dest_field, added_fields|
       # When multiple calls, added_field is an array
+
+      dest_field_value = event[dest_field]
+
       Array(added_fields).each do |added_field|
-        if event[dest_field].is_a?(Hash) ^ event[added_field].is_a?(Hash)
+        added_field_value = event[added_field]
+
+        dest_field_value_is_a_hash = dest_field_value.is_a?(Hash)
```

Comment:
I wouldn't DRY this just for 2 repetitions, the performance impact is very small and IMO it hurts readability